Fluid/Ast/IdentifierSegment.cs

```diff
@@ -36,8 +38,17 @@ public override ValueTask<FluidValue> ResolveAsync(Scope value, TemplateContext
             if (result.IsNil() && context.Model != null)
             {
                 // Check for a custom registration
-                _accessor ??= context.Options.MemberAccessStrategy.GetAccessor(context.Model.GetType(), Identifier);
-                _accessor ??= MemberAccessStrategyExtensions.GetNamedAccessor(context.Model.GetType(), Identifier, MemberNameStrategies.Default);
+                var modelType = context.Model.GetType();
+                if (modelType != _type)
```

Comment:
I checked the type first to avoid extra computation for the accessor, this way the accessor will be computed if the model type has been changed